analytic_engine/src/sst/parquet/hybrid.rs

```diff
@@ -0,0 +1,435 @@
+            field_types.push(IndexedKind {
+                idx,
+                kind: schema.column(idx).data_type,
+            });
+        }
+    }
+    // TODO: should keep tsid ordering here?
+    let mut batch_by_tsid = BTreeMap::new();
+    for record_batch in arrow_record_batch_vec {
+        let tsid_array = record_batch
+            .column(tsid_type.idx)
+            .as_any()
+            .downcast_ref::<UInt64Array>()
+            .expect("checked when create table");
+
+        if tsid_array.is_empty() {
+            continue;
+        }
+
+        let tagv_columns = tag_types
+            .iter()
+            .map(|indexed_name| {
+                record_batch
+                    .column(indexed_name.idx)
+                    .as_any()
+                    .downcast_ref::<StringArray>()
+                    .expect("checked when create table")
+            })
+            .collect::<Vec<_>>();
+        let mut previous_tsid = tsid_array.value(0);
+        // duplicated_tsids is an array of every tsid's offset in origin array
+        // the length of each tsid can be calculated with
+        // tsid_n = duplicated_tsids[n+1].offset - duplicated_tsids[n].offset
+        let mut duplicated_tsids = vec![(previous_tsid, 0)]; // (tsid, offset)
+        for row_idx in 1..tsid_array.len() {
+            let tsid = tsid_array.value(row_idx);
+            if tsid != previous_tsid {
+                previous_tsid = tsid;
+                duplicated_tsids.push((tsid, row_idx));
+            }
+        }
+        for i in 0..duplicated_tsids.len() {
+            let (tsid, offset) = duplicated_tsids[i];
+            let length = if i == duplicated_tsids.len() - 1 {
+                tsid_array.len() - offset
+            } else {
+                duplicated_tsids[i + 1].1 - offset
+            };
+
+            let mut field_columns = Vec::with_capacity(field_types.len());
+            for indexed_type in &field_types {
+                let fields_in_one_tsid =
+                    record_batch.column(indexed_type.idx).slice(offset, length);
+                field_columns.push(fields_in_one_tsid)
+            }
+            let batch = batch_by_tsid.entry(tsid).or_insert_with(|| {
+                TsidBatch::new(
+                    tagv_columns
+                        .iter()
+                        .map(|col| col.value(offset).to_string())
+                        .collect(),
+                    record_batch.column(timestamp_type.idx).clone(),
+                    field_types
+                        .iter()
+                        .map(|indexed_name| record_batch.column(indexed_name.idx).clone())
+                        .collect(),
+                )
+            });
+            batch.append_postition(offset, length);
+        }
+    }
+
+    build_hybrid_record(
+        hybrid_arrow_schema,
+        tsid_type,
+        timestamp_type,
+        tag_types,
+        field_types,
+        batch_by_tsid,
+    )
+}
+
+#[cfg(test)]
```

Comment:
The coverage looks not enough to me. Can we add more tests for this? Like round-trip (encode and decode) all supported types or Arrow arrays, and parqute-related tests